app/modules/examinations/router_v2.py

```
@@ -83,6 +130,113 @@ def delete_examination(exam_id: int, db: Session = Depends(get_db), principal: P
+        "SELECT id, exam_id, subject_id, class_id, teacher_id, total_marks "
+        "FROM exam_subjects WHERE school_id = :school_id AND exam_id = :exam_id "
+        "ORDER BY class_id, subject_id"
+    ), {"school_id": principal.school_id, "exam_id": exam_id}).mappings().all()
+    return [dict(row) for row in rows]
+
+
+@router.post("/examinations/{exam_id}/subjects", status_code=201)
+def assign_exam_subject(
+    exam_id: int,
+    payload: ExamSubjectAssignment,
+    db: Session = Depends(get_db),
```

Ruff (0.16.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)